src/main/java/hudson/plugins/ec2/win/WinConnection.java

```diff
@@ -124,11 +131,13 @@ public boolean ping() {
     
     public boolean pingFailingIfSSHHandShakeError() throws IOException {
         LOGGER.log(Level.FINE, () -> "checking SMB connection to " + host);
-        try (Socket socket = new Socket()) {
-            socket.connect(new InetSocketAddress(host, 445), TIMEOUT);
-            winrm().ping();
+        try (
+            Socket socket = new Socket();
             Connection connection = smbclient.connect(host);
             Session session = connection.authenticate(authentication);
+        ) {
```

Comment:
### Second socket leak
Fix: moved `connection` and  `session` to `try with resources` block so that they get closed properly.